developer_docs/ansible_dev_setup.asciidoc

```diff
@@ -0,0 +1,138 @@
+= Ansible devel setup
+:toc: right
+:toclevels: 5
+
+[[prerequisites]]
+== Prerequisites
+
+* https://github.com/theforeman/foreman/blob/develop/developer_docs/foreman_dev_setup.asciidoc[Foreman devel setup]
+* https://github.com/theforeman/foreman/blob/develop/developer_docs/smart_proxy_dev_setup.asciidoc[Smart Proxy devel setup]
+* https://github.com/theforeman/foreman/blob/develop/developer_docs/remote_execution_dev_setup.asciidoc[Remote Execution devel setup]
+* Installed https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html[Ansible]
+* Running machine that can be connected to via SSH (key authentication is best)
+
+[[foreman-plugin]]
+== Foreman plugin
+
+Download https://github.com/theforeman/foreman_ansible[foreman_ansible] plugin
+[source, bash]
+....
+git clone https://github.com/theforeman/foreman_ansible.git
+....
+
+Enable `foreman-ansible` plugin
+[source, ruby]
+....
+# bundler.d/ansible.local.rb
+
+gem 'foreman_ansible', path: '/path/to/foreman_ansible'
+....
+
+Install plugin dependencies & migrate database
+[source, bash]
+....
+bundle install
+npm install
+
+bundle exec rails db:migrate
+....
+
+
+[[smart-proxy-plugin]]
+== Smart Proxy plugin
+Add plugin gems
+
+[source, bash]
+....
+# /path/to/smart-proxy/bundler.d/ansible.local.rb
+
+gem 'smart_proxy_ansible', git: 'https://github.com/theforeman/smart_proxy_ansible.git'
+....
+
+Install gems
+[source, bash]
+....
+bundle install
+....
+
+Enable feature
+
+[source, yaml]
+....
+# config/settings.d/ansible.yml
+---
+:enabled: true
+....
+
+Run all servicies
+[source, bash]
+....
+bundle exec foreman start
+bundle exec rake dynflow:executor
+
+bundle exec bin/smart-proxy
+# => [I] Successfully initialized 'ansible'
+....
+
+[[foreman-collection]]
+== Foreman collection
+
+`theforeman.foreman` collection contains Ansible Modules to manage Foreman and Katello installations.
+
+Install https://docs.ansible.com/ansible/latest/collections/theforeman/foreman/index.html[Foreman Collection]
+
+[source, bash]
+....
+ansible-galaxy collection install theforeman.foreman
+....
+
+Create symbolic link to folder where you have your collections
+
+[source, bash]
+....
+sudo ln -s /home/<your-account>/.ansible/collections/ansible_collections/ /usr/share/ansible/collections/ansible_collections
```

Comment:
I'd prefer if we could avoid writing stuff into system directories

Reply:
So what would be the proper fix? Configure ansible to store collections somewhere else?

Reply:
The whole callback configuration can be done without writing to any system paths

```
# Install the collection
# ansible-galaxy install theforeman.foreman
Process install dependency map  
Starting collection install process  
Installing 'theforeman.foreman:3.5.0' to '/home/aruzicka/.ansible/collections/ansible_collections/theforeman/foreman'

# Copy the default configuration to user's HOME
# cp /etc/ansible/ansible.cfg ~/.ansible.cfg

# Add theforeman.foreman.foreman callback plugin to a callback plugin whitelist
# ansible -i ,localhost -c local -m ini_file -a 'path=~/.ansible.cfg state=present section=defaults option=callback_whitelist value=theforeman.foreman.foreman' localhost

# Set url for the callback plugin
# ansible -i ,localhost -c local -m ini_file -a 'path=~/.ansible.cfg state=present section=callback_foreman option=url value=http://localhost:3000' localhost
```